auth/passwords.hh

```diff
@@ -84,6 +84,7 @@ sstring generate_salt(RandomNumberEngine& g, scheme scheme) {
 
 ///
 /// Async version of `hash_with_salt` that returns a future.
+/// If possible, hashing uses `coroutine::maybe_yield` to prevent reactor stalls.
```

Comment:
Why "if possible"? You do use it. I would say something like:
This function avoids stalls by yielding during long computations.
(or use whatever wording you prefer).